src/savers/tvg/tvgTvgSaver.cpp

```diff
@@ -0,0 +1,211 @@
+    outFile.write(buffer,size);
+    outFile.close();
+
+    if (buffer) free(buffer);
+    buffer = nullptr;
+    bufferPosition = nullptr;
+    size = 0;
+    reserved = 0;
+};
+
+
+bool TvgSaver::close()
+{
+    this->done();
+
+    if (buffer) free(buffer);
+    buffer = nullptr;
+    bufferPosition = nullptr;
+    size = 0;
+    reserved = 0;
+
+    return true;
+}
+
+
+void TvgSaver::saveMemberIndicator(TvgIndicator ind)
+{
+    if (size + sizeof(TvgIndicator) > reserved) resizeBuffer(size + sizeof(TvgIndicator));
+
+    memcpy(bufferPosition, &ind, sizeof(TvgIndicator));
+    bufferPosition += sizeof(TvgIndicator);
+    size += sizeof(TvgIndicator);
+}
+
+
+void TvgSaver::saveMemberDataSize(ByteCounter byteCnt)
+{
+    if (size + sizeof(ByteCounter) > reserved) resizeBuffer(size + sizeof(ByteCounter));
+
+    memcpy(bufferPosition, &byteCnt, sizeof(ByteCounter));
+    bufferPosition += sizeof(ByteCounter);
+    size += sizeof(ByteCounter);
+}
+
+
+void TvgSaver::saveMemberDataSizeAt(ByteCounter byteCnt)
+{
+    memcpy(bufferPosition - byteCnt - sizeof(ByteCounter), &byteCnt, sizeof(ByteCounter));
+}
+
+
+void TvgSaver::skipMemberDataSize()
+{
+    if (size + sizeof(ByteCounter) > reserved) resizeBuffer(size + sizeof(ByteCounter));
+    bufferPosition += sizeof(ByteCounter);
+    size += sizeof(ByteCounter);
+}
+
+
+ByteCounter TvgSaver::saveMemberData(const void* data, ByteCounter byteCnt)
+{
+    if (size + byteCnt > reserved) resizeBuffer(size + byteCnt);
+
+    memcpy(bufferPosition, data, byteCnt);
+    bufferPosition += byteCnt;
+    size += byteCnt;
+
+    return byteCnt;
+}
+
+
+ByteCounter TvgSaver::saveMember(TvgIndicator ind, ByteCounter byteCnt, const void* data)
+{
+    ByteCounter blockByteCnt = sizeof(TvgIndicator) + sizeof(ByteCounter) + byteCnt;
+
+    if (size + blockByteCnt > reserved) resizeBuffer(size + blockByteCnt);
+
+    memcpy(bufferPosition, &ind, sizeof(TvgIndicator));
+    bufferPosition += sizeof(TvgIndicator);
+    memcpy(bufferPosition, &byteCnt, sizeof(ByteCounter));
+    bufferPosition += sizeof(ByteCounter);
+    memcpy(bufferPosition, data, byteCnt);
+    bufferPosition += byteCnt;
+
+    size += blockByteCnt;
+
+    return blockByteCnt;
+}
+
+
+void TvgSaver::resizeBuffer(uint32_t newSize)
+{
+    // MGS - find more optimal alg ?
+    reserved += 100;
+    if (newSize > reserved) reserved += newSize - reserved + 100;
+
+    auto bufferOld = buffer;
+
+    buffer = static_cast<char*>(realloc(buffer, reserved));
+
+    if (buffer != bufferOld)
+        bufferPosition = buffer + (bufferPosition - bufferOld);
+}
+
+
+void TvgSaver::rewindBuffer(ByteCounter bytesNum)
+{
+    if (bufferPosition - bytesNum < buffer) return;
+
+    bufferPosition -= bytesNum;
+    size -= bytesNum;
+}
+
+
+bool TvgSaver::saveHeader()
+{
+    const char *tvg = TVG_HEADER_TVG_SIGN_CODE;
+    const char *version = TVG_HEADER_TVG_VERSION_CODE;
+    // MGS - metadata not raelly used for now - type hardcoded (to be changed)
```

Comment:
if it's not used, remove it. we can improve this by adding checkSum mechanism later.